modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java

```diff
@@ -117,36 +121,20 @@ public void visitObject(ObjectType objectType)
                 .map(ImportedTypesModelProperty::getImportedTypes);
         importedTypes = typeImports.isPresent() ? typeImports.get() : ImmutableMap.of();
 
-        extensionModel.getConfigurationModels().stream()
-                .forEach(config -> validateParameters(config.getParameterModels(), visitor, config.getName(),
-                                                      CONFIGURATION, extensionModel.getName()));
-
-        validateOperations(extensionModel, extensionModel.getOperationModels(), visitor);
-        extensionModel.getConfigurationModels().forEach(config -> validateOperations(extensionModel, config.getOperationModels(), visitor));
-
-        validateConnectionProviders(extensionModel, extensionModel.getConnectionProviders(), visitor);
-        extensionModel.getConfigurationModels().forEach(config -> validateConnectionProviders(extensionModel, config.getConnectionProviders(), visitor));
-    }
-
-    private void validateConnectionProviders(ExtensionModel extensionModel, List<ConnectionProviderModel> providers, MetadataTypeVisitor visitor)
-    {
-        providers.forEach(provider -> validateParameters(provider.getParameterModels(), visitor, provider.getName(),
-                                                         CONNECTION_PROVIDER, extensionModel.getName()));
-    }
-
-    private void validateOperations(ExtensionModel extensionModel, List<OperationModel> operations, MetadataTypeVisitor visitor)
-    {
-        operations.forEach(operation -> validateParameters(operation.getParameterModels(), visitor, operation.getName(),
-                                                           OPERATION, extensionModel.getName()));
-    }
-
-    private void validateParameters(List<ParameterModel> parameters, MetadataTypeVisitor visitor, String ownerName, String ownerModelType, String extensionName)
-    {
-        parameters.stream().forEach(parameterModel -> {
-            validateParameter(parameterModel, visitor, ownerName, ownerModelType, extensionName);
-            validateNameCollisionWithTypes(parameterModel, ownerName, ownerModelType, extensionName,
-                                           parameters.stream().map(p -> hyphenize(p.getName())).collect(toList()));
-        });
+        new ExtensionWalker()
+        {
+            @Override
+            public void onParameter(ParameterizedModel owner, ParameterModel model)
+            {
+                String ownerName = owner.getName();
+                String ownerModelType = owner.getClass().getSimpleName();
```

Comment:
We used to have an specific tag (CONNECTION_PROVIDER, OPERATION, CONFIGURATION, etc) to use in the error message,  but now with the ExtensionWalker it is no longer possible to identify what kind of component is owner. Any idea?